src/gt_frontend/gtscript_ast.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+#
+# Eve Toolchain - GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2020, CSCS - Swiss National Supercomputing Center, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from typing import List, Union
+
+import gtc.common as common
+from eve import Node
+
+
+__all__ = [
+    "GTScriptAstNode",
+    "Statement",
+    "Expr",
+    "Symbol",
+    "IterationOrder",
+    "Constant",
+    "Interval",
+    "LocationSpecification",
+    "SubscriptSingle",
+    "SubscriptMultiple",
+    "BinaryOp",
+    "Call",
+    "LocationComprehension",
+    "Generator",
+    "Assign",
+    "Stencil",
+    "Pass",
+    "Argument",
+    "Computation",
+]
+
+
+class GTScriptAstNode(Node):
+    pass
+
+
+class Statement(GTScriptAstNode):
+    pass
+
+
+class Expr(GTScriptAstNode):
+    pass
+
+
+class Symbol(Expr):
+    name: str
+
+
+class IterationOrder(GTScriptAstNode):
+    order: str
+
+
+# todo: use type parameter see https://github.com/samuelcolvin/pydantic/pull/595
+# T = TypeVar('T')
+# class Constant(GT4PyAstNode, Generic[T]):
+#    value: T
+
+
+class Constant(Expr):
+    # todo: use StrictStr, StrictInt, StrictFloat as pydantic automatically converts into the first
```

Comment:
Converting ints to floats looks like an important issue to me...

Reply:
In principle yes, but for now it's fine. When you get write an `int` it gets converted to a `float`, which doesn't matter until the `int` is sufficiently large in magnitude. Left as a todo.